subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -930,13 +952,31 @@ int nrf_cloud_rest_agps_data_get(struct nrf_cloud_rest_context *const rest_ctx,
 		goto clean_up;
 	}
 
+/** In filtered ephemeris mode, request A-GPS data no more often than
+ *  every 2 hours (time in milliseconds). Without this, the GNSS unit will
+ *  request for ephemeris every hour because a full set was not received.
+ */
+#define MARGIN_MINUTES 10
+#define AGPS_UPDATE_PERIOD ((120 - MARGIN_MINUTES) * 60 * MSEC_PER_SEC)
+
+	if (filtered && (last_request_timestamp != 0) &&
```

Comment:
This check makes testing with modem_shell difficult, because A-GPS downloads (with filtering enabled) can only be done once every two hours. But I guess there's no easy way around this.

Reply:
I could make AGPS_UPDATE_PERIOD a config variable.  If it were set to 0, it would never skip the download.

Reply:
I propose that we keep it like this for now. I'm not sure if we should add a new configuration option just to make testing more easier. The normal case can be run using modem_shell with the current implementation. If the user wants to fetch filtered ephemerides again before 2 hours have passed, the device just has to be rebooted.